tests/benchmark/compute/instruction/test_storage.py

```diff
@@ -346,11 +349,23 @@ def test_storage_access_warm(
     contract_address = compute_create_address(address=sender_addr, nonce=0)
 
     with TestPhaseManager.execution():
-        op_tx = Transaction(
-            to=contract_address,
-            gas_limit=gas_benchmark_value,
-            sender=pre.fund_eoa(),
+        tx_max_gas_limit = (
+            min(gas_benchmark_value, tx_gas_limit_cap)
+            if tx_gas_limit_cap is not None
+            else gas_benchmark_value
         )
-        blocks.append(Block(txs=[op_tx]))
+        num_exec_txs = math.ceil(gas_benchmark_value / tx_max_gas_limit)
+        txs = []
+        for i in range(num_exec_txs):
+            tx_gas_limit = min(
+                tx_max_gas_limit, gas_benchmark_value - i * tx_max_gas_limit
+            )
+            op_tx = Transaction(
+                to=contract_address,
+                gas_limit=tx_gas_limit,
+                sender=pre.fund_eoa(),
+            )
+            txs.append(op_tx)
+        blocks.append(Block(txs=txs))
```

Comment:
```suggestion
        num_exec_txs = math.ceil(gas_benchmark_value / tx_gas_limit)
        txs = []
        for i in range(num_exec_txs):
            gas_limit = min(
                tx_max_gas_limit, gas_benchmark_value - i * tx_gas_limit
            )
            op_tx = Transaction(
                to=contract_address,
                gas_limit=gas_limit,
                sender=pre.fund_eoa(),
            )
            txs.append(op_tx)
        blocks.append(Block(txs=txs))
```

According to the previous comments